docs/source/design/python_imports.rst

```diff
@@ -0,0 +1,65 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_python_imports:
+
+************************
+Python import statements
+************************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current (albeit incomplete) implementation.
+
+The following rules apply to any ``import`` statement in the code base:
+
+- All imports *must* be absolute, unless they import individual pieces of an integrated code component that is only split across several source code files for technical or organizational reasons.
+
+- Imports *must* be placed at the top of a source file, unless there is a
+  specific reason not to do so (e.g., delayed import due to performance
+  concerns, circular dependencies). If such a reason exists, it *must*
+  be documented by a comment at the import statement.
+
+- There *must* be no more than one import per line.
+
+- Multiple individual imports from a single module *must* follow the pattern::
+
+      from <module> import (
+          symbol1,
+          symbol2,
+      )
+
+  Individual imported symbols *should* be sorted alphabetically. The last symbol
+  line *should* end with a comma.
+
+- Imports from packages and modules *should* be grouped in categories like
+
+  - Standard library packages
+
+  - 3rd-party packages
+
+  - Datalad core (absolute imports)
+
+  - Datalad extensions
+  
+  - Datalad core ("local" relative imports)
+  
+  Sorting imports can be aided by https://github.com/PyCQA/isort (e.g. ``python -m isort -m3 -fgw 2 -tc <filename>``).
```

Comment:
`--fgw` and `--tc` need to be written with two hyphens.

Reply:
```suggestion
  Sorting imports can be aided by https://github.com/PyCQA/isort (e.g. ``python -m isort -m3 --fgw 2 --tc <filename>``).
```

Reply:
FWIW if #5923 is merged (I think we can do it since it should not impose anything on anyone), it could just be

```suggestion
  Sorting imports can be aided by https://github.com/PyCQA/isort (e.g. ``python -m isort <filename>``).
```
and we could (later) recommend installing/enabling `pre-commit` to automate that

Reply:
I'd say, let's move with this one now, and adjust later as necessary and possible.